### PR TITLE
Addressing issue 2747, login appearing in scratchbook window

### PR DESCRIPTION
**Pull request by sszakony**
To fix this problem, I added a new option to the Login and User Registration options called "noscratchbook", which when true prevents an action from opening in a scratchbook window rather than in the Galaxy central pane. The "noscratchbook" parameter is defaulted to false, so it should not interfere with the handling of any dialog that does not include this option explicitly. 

Once the new option was implemented, I added a test in scratchbook.js so that when the test is made to see if scratchbook is enabled, an additional check is done on this option parameter, and if the parameter is true the availability of scratchbook is ignored and the action opens in the central pane.

I also tested visualization options to ensure that they would not be adversely affected. The value received for the "noscratchbook" option in these cases is "undefined", which evaluates as a false in the scratchbook.js test.

### CODE REVIEW

**Comment by jmchilton** [comment]
@galaxybot test this

**Comment by jmchilton** [comment]
Ping @guerler

File: database/info.txt [hunk 1]
```diff
@@ -1 +0,0 @@
-This folder contains the data
```
Comment (nsoranzo) [inline database/info.txt:1]: With this change the database/ folder is no more created automatically on a fresh clone, is this safe?

Comment (nsoranzo) [reply database/info.txt:1]: Indeed at least Travis startup tests are broken.

**Pull request merged**